scripts/release/homebrew/docker/formula_generate.py

```diff
@@ -134,7 +133,15 @@ def update_formula() -> str:
     line_idx_to_remove = set()
     upgrade = False
     for idx, line in enumerate(lines):
-        if line.strip().startswith("resource"):
+        # In released formula, the url is in the release tag format, such as
+        # "https://github.com/Azure/azure-cli/archive/azure-cli-2.17.1.tar.gz".
+        # version is extracted from url. During build, the url is in the format like
+        # "https://codeload.github.com/Azure/azure-cli/legacy.tar.gz/7e09fd50c9ef02e1ed7d4709c7ab1a71acd3840b".
```

Comment:
We can use the new API to download the source code:

https://github.com/Azure/azure-cli/archive/7e09fd50c9ef02e1ed7d4709c7ab1a71acd3840b.zip

`7e09fd50c9ef02e1ed7d4709c7ab1a71acd3840b` is the commit ID 7e09fd50c9ef02e1ed7d4709c7ab1a71acd3840b.